src/main/scala/com/campudus/tableaux/database/model/FileModel.scala

```diff
@@ -154,10 +159,10 @@ class FileModel(override protected[this] val connection: DatabaseConnection) ext
           |filename,
           |idfolder,
           |created_at,
-          |updated_at FROM $table WHERE idfolder = ? AND tmp = FALSE""".stripMargin
+          |updated_at FROM $table WHERE ${condition._1} AND tmp = FALSE""".stripMargin
 
     for {
-      result <- connection.query(select, Json.arr(folder))
+      result <- connection.query(select, condition._2)
```

Comment:
This might trigger a `PreparedStatement`.

What about this?
```
def select(condition: String) = s"""...your select statement with ${condition}..."""

folder match {
  case Some(id) => connection.query(select("idfolder = ?"), JsonArr(id.toString))
  case None => connection.query(select("idfolder = null"))
}
```